libmscore/slur.cpp

```diff
@@ -910,76 +913,115 @@ void Slur::slurPos(SlurPos* sp)
             xo = hw1 * .5;
             if (note1)
                   yo = note1->pos().y();
-            else if(_up)
-                  yo = startCR()->bbox().top();
+            else if (_up)
+                  yo = scr->bbox().top();
             else
-                  yo = startCR()->bbox().top() + startCR()->height();
+                  yo = scr->bbox().top() + scr->height();
             yo += _spatium * .9 * __up;
 
+            // adjustments for stem and/or beam
+
             if (stem1) { //sc not null
                   Beam* beam1 = sc->beam();
                   if (beam1 && (beam1->elements().back() != sc) && (sc->up() == _up)) {
+                        // start chord is beamed but not the last chord of beam group
+                        // and slur direction is same as start chord (stem side)
+
+                        // in these cases, layout start of slur to stem
+
                         qreal sh = stem1->height() + _spatium;
                         if (_up)
                               yo = sc->downNote()->pos().y() - sh;
                         else
                               yo = sc->upNote()->pos().y() + sh;
                         xo       = stem1->pos().x();
-                        stemPos  = true;
+
+                        // force end of slur to layout to stem as well,
+                        // if start and end chords have same stem direction
+                        stemPos = true;
                         }
                   else {
+                        // start chord is not beamed or is last chord of beam group
+                        // or slur direction is opposite that of start chord
+
+                        // at this point slur is in default position relative to note on slur side
+                        // but we may need to make further adjustments
+
+                        // if stem and slur are both up
+                        // we need to clear stem horizontally
                         if (sc->up() && _up)
                               xo = hw1 + _spatium * .3;
+
                         //
                         // handle case: stem up   - stem down
                         //              stem down - stem up
                         //
                         if ((sc->up() != ecr->up()) && (sc->up() == _up)) {
-                              Note* n1  = sc->up() ? sc->downNote() : sc->upNote();
+                              // start and end chord have opposite direction
+                              // and slur direction is same as start chord
+                              // (so slur starts on stem side)
+
+                              // float the start point along the stem to follow direction of movement
+                              // see for example Gould p. 111
+
+                              // get position of note on slur side for start & end chords
+                              Note* n1  = sc->up() ? sc->upNote() : sc->downNote();
```

Comment:
Well, Finale is doing it wrong according to Gould.  The outer slur is from E to E and should thus be horizontal.  The inner slur is from B to B and should also be horizontal.  Previously we were getting the inner case right but not the outer, because we were calculating the amount of exposed stem based on the wrong note.  In fact, if you take that example and then take the "F" at the end of the first measure and transpose it down an octave, you'll  see the slur fly upward, before my fix.